File: flutter_client/lib/models/user_model.dart
class UserModel {
  final String id;
  final String name;
  final String title;
  final String email;
  final int age;
  final String mobileNo;

  UserModel({
    required this.id,
    required this.name,
    required this.title,
    required this.email,
    required this.age,
    required this.mobileNo,
  });

  //Convert Dart Object to Json Data
  Map<String, dynamic> toJson() {
    return {"name": name, "title": title, "email": email, "age": age, "mobileNo": mobileNo};
  }

  //Convert Json Data to Dart Object
  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      id: json["_id"],
      name: json["name"],
      title: json["title"],
      email: json["email"],
      age: json["age"],
      mobileNo: json["mobileNo"],
    );
  }
}

File: node_backend/models/userModel.js
const mongoose = require("mongoose");

const userSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
        trim: true,
    },
    title: {
        type: String,
        required: true,
        trim: true,
    },
    email: {
        type: String,
        required: true,
        trim: true,
        unique: true,
    },
    age: {
        type: Number,
        required: true,
    },
    mobileNo: {
        type: String,
        required: true,
        trim: true,
    }
},
    { timestamps: true }

);

const User = mongoose.model("User", userSchema);
module.exports = User;